processing/src/main/java/org/apache/druid/segment/nested/NestedDataColumnSupplier.java

```diff
@@ -91,6 +95,15 @@ public NestedDataColumnSupplier(
         fields = GenericIndexed.read(bb, GenericIndexed.STRING_STRATEGY, mapper);
         fieldInfo = NestedLiteralTypeInfo.read(bb, fields.size());
 
+        if (fields.size() == 1 &&
+            ((version == 0x03 && NestedPathFinder.JQ_PATH_ROOT.equals(fields.get(0))) ||
+             (version == 0x04 && NestedPathFinder.JSON_PATH_ROOT.equals(fields.get(0))))
+        ) {
+          simpleType = fieldInfo.getTypes(0).getSingleType();
+        } else {
+          simpleType = null;
+        }
```

Comment:
Just generally speaking, all of this work is being done in a constructor.  It would be better to have it in a static method and just pass the needed things into the constructor.  I realize it's a nit comment on old code...

Reply:
agree, will rework this in a follow-up PR